custom_components/tuya_local/devices/truelife_air_purrifier_p3.yaml

```diff
@@ -0,0 +1,118 @@
+name: Air Purifier
+products:
+  - id: pc3mcvhmyfponsqf
+    manufacturer: "TrueLife"
+    model: "Air Purifier P3 WIFI"
+entities:
+  - entity: fan
+    icon: "mdi:air-purifier"
+    translation_only_key: fan_with_presets
+    dps:
+      - id: 1
+        type: boolean
+        name: switch
+      - id: 3
+        name: preset_mode
+        type: string
+        mapping:
+          - dps_val: AUTO
+            value: smart
+          - dps_val: SLEEP
+            value: sleep
+          - dps_val: MANUAL
+            value: manual
+  - entity: select
```

Comment:
The fan speed can be merged to the fan entity, with mappings to integers in the 0-100 range.